### DISCUSSION TITLE
Theming for the home (~) directory

### DISCUSSION THREAD

**Discussion by fdnt7** [sxyazi/yazi#1077] [1. Q&A] [answered]
I wanted to know what matcher pattern to use in the `name` parameter when theming the home (`~`) directory itself (not `/home`). I've tried `"~"` but it didn't work. Could it be that this is not currently possible? Thank you in advance!
`yazi` version: [58e5d22](https://github.com/sxyazi/yazi/commit/58e5d2280ab218f5d35be245cbaefb3da3b76d66)
This is a great project by the way, keep up the great work :)

**Comment by sxyazi** [comment]
Could you post your config snippets so I can see what might be causing the issue?

**Reply by fdnt7** [reply]
To clarify, doing `*/home/fdnt/` and `/home/fdnt/` (`fdnt` is my username) both works, I was just asking if it's possible to just use a generic `~` that will just resolve to `/home/<username>` automatically without hard coding the username in the `theme.toml`, and so the same config can be used for other users with different names.
I've tried `~`, `~/"`, `*/~/`. None of them seems to work. I also don't think there's anything wrong with my `theme.toml`, but [here](https://github.com/fdnt7/nix-config/blob/034da45ae6bf74a47fd6f8e57bf8536c1aa344fc/home-manager/yazi/theme.toml#L529-L532) is it just for good measure.

**Reply by lpnh** [reply]
hey, hi! how about something like this:

```
[icon]
rules = [{ name = "/home/*/", text = "icon" }]
```

edit: Sorry, it didn't work as I thought it would.

**Comment by sxyazi** [comment, marked as answer]
You can achieve it by overriding the [`Entity:icon()` method](https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/components/entity.lua). Add the following to your `init.lua`:

```lua
local old_entity_icon = Entity.icon
function Entity:icon()
  if self._file.url == Url(os.getenv("HOME")) then
    return ui.Line(" [icon] ")
  end
  return old_entity_icon(self)
end
```

![screenshot-000055](https://github.com/sxyazi/yazi/assets/17523360/09aaa8e1-36f7-4961-abb8-1df06fffbeda)
